Source/DafnyCore/AST/DafnyAst.cs

```diff
@@ -77,108 +78,114 @@ public abstract class INode {
+              if (token1.pos < startTok.pos) {
+                startTok = token1;
+              }
 
-            if (token1.pos < startTok.pos) {
-              startTok = token1;
+              if (token1.pos + token1.val.Length > endTok.pos + endTok.val.Length) {
+                endTok = token1;
+              }
             }
 
-            if (token1.pos + token1.val.Length > endTok.pos + endTok.val.Length) {
-              endTok = token1;
+            void updateStartEndTokRecursive(INode node) {
+              if (node is null) {
+                return;
+              }
+
+              if (node.tok.Filename != tok.Filename || node is Expression { IsImplicit: true } ||
+                  node is DefaultValueExpression) {
+                // Ignore any auto-generated expressions.
+              } else if (node != this && node.RangeToken != null) {
+                UpdateStartEndToken(node.StartToken);
+                UpdateStartEndToken(node.EndToken);
+              } else {
+                UpdateStartEndToken(node.tok);
+              }
+
+              node.Children.Iter(updateStartEndTokRecursive);
             }
-          }
 
-          void UpdateStartEndTokRecursive(INode node) {
-            if (node is null) {
-              return;
-            }
+            updateStartEndTokRecursive(this);
 
-            if (node.tok.Filename != tok.Filename || node is Expression { IsImplicit: true } ||
-                node is DefaultValueExpression) {
-              // Ignore any auto-generated expressions.
-            } else if (node != this && node.GetRangeToken() != null) {
-              UpdateStartEndToken(node.GetStartToken());
-              UpdateStartEndToken(node.GetEndToken());
-            } else {
-              UpdateStartEndToken(node.tok);
+            if (FormatTokens != null) {
+              foreach (var token in FormatTokens) {
+                UpdateStartEndToken(token);
+              }
             }
 
-            node.Children.Iter(UpdateStartEndTokRecursive);
-          }
-
-          UpdateStartEndTokRecursive(this);
-
-          if (FormatTokens != null) {
-            foreach (var token in FormatTokens) {
-              UpdateStartEndToken(token);
-            }
+            rangeToken = new RangeToken(startTok, endTok);
           }
+        }
 
-          rangeToken = new RangeToken(startTok, endTok);
+        if (rangeToken.Filename == null) {
+          rangeToken.Filename = tok.Filename;
         }
-      }
 
-      if (rangeToken.Filename == null) {
-        rangeToken.Filename = tok.Filename;
+        return rangeToken;
       }
-
-      return rangeToken;
+      set => rangeToken = value;
     }
 
-    public IToken GetStartToken() => GetRangeToken()?.StartToken;
-    public IToken GetEndToken() => GetRangeToken()?.EndToken;
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+    public IToken StartToken => RangeToken?.StartToken;
+
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
```

Comment:
Why do we need this attribute?

Reply:
For now, to avoid recomputing StartToken / RangeToken in the debugger view, which you said could cause problems.

Reply:
When I had trouble with it the computation of Token.ToString() was still broken, so let's try not adding this attribute.